src/omnisharp/OmnisharpDownloader.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { PackageManager, Package, Status } from '../packages';
+import { PlatformInformation } from '../platform';
+import { Logger } from '../logger';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { GetPackagesFromVersion, GetVersionFilePackage } from './OmnisharpPackageCreator';
+import { SetStatus, LogPlatformInformation, ReportInstallationError, SendInstallationTelemetry, GetNetworkDependencies } from '../OmnisharpDownload.Helper';
+
+export class OmnisharpDownloader {
+    private status: Status;
+    private statusItem: vscode.StatusBarItem;
+    private proxy: string;
+    private strictSSL: boolean;
+    private packageManager: PackageManager;
+    private telemetryProps: any;
+
+    public constructor(
+        private channel: vscode.OutputChannel,
+        private logger: Logger,
+        private packageJSON: any,
+        private platformInfo: PlatformInformation,
+        private reporter?: TelemetryReporter) {
+
+        let statusObject = SetStatus();
+        this.status = statusObject.Status;
+        this.statusItem = statusObject.StatusItem;
+
+        let networkObject = GetNetworkDependencies();
+        this.proxy = networkObject.Proxy;
+        this.strictSSL = networkObject.StrictSSL;
+
+        this.telemetryProps = {};
+        this.packageManager = new PackageManager(this.platformInfo, this.packageJSON);
+    }
+
+    public async DownloadAndInstallOmnisharp(version: string, serverUrl: string, installPath: string) {
+        if (!version) {
+            throw new Error('Invalid version');
```

Comment:
If this happens, this is basically a bug in our code, right? Since this method is not expecting a null version. "invalid version" makes it sound like a user mistake, which it isn't.